src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -709,13 +727,32 @@ private static Expression BindParameterFromRouteValueOrQueryString(ParameterInfo
             return BindParameterFromValue(parameter, Expression.Coalesce(routeValue, queryValue), factoryContext);
         }
 
+        private static Expression BindParameterFromFormFile(ParameterInfo parameter, string key, FactoryContext factoryContext)
+        {
+            if (factoryContext.JsonRequestBodyType is not null)
```

Comment:
I would add a computed property on the factoryContext `HasBody => JsonRequestBodyType is not null || ReadForm` and use that here and in the JSON logic

Reply:
The problem with doing exactly that is it wouldn't allow you to have more than one `IFormFile` parameter as trying to bind the second one would find the first and then throw.